### PR TITLE
Generic ocs method

### PR DESCRIPTION
**Pull request by nickvergessen**
@PVince81 as discussed

### CODE REVIEW

**Comment by PVince81** [comment]
Do you mind adding a test case as well ?

**Comment by PVince81** [comment]
The code looks good so far

File: owncloud/owncloud.py [hunk 1]
```diff
@@ -1396,6 +1408,26 @@ def __check_ocs_status(tree, accepted_codes=[100]):
...
+        """Makes a OCS API request and analyses the response
+
+        :param method: HTTP method
+        :param service: service name
+        :param action: action path
+        :param \*\*kwargs: optional arguments that ``requests.Request.request`` accepts
+        :returns :class:`requests.Response` instance
+        """
+
+        accepted_codes = kwargs.get('accepted_codes', [100])
```
Comment (nickvergessen) [inline owncloud/owncloud.py]: needs to be removed from kwargs

```
CRITICAL - owner_worker - Exception occured: request() got an unexpected keyword argument 'accepted_codes' 
 Traceback (most recent call last):
  File "./python/smashbox/multiprocessing_engine.py", line 164, in worker_wrap
    f(step)
  File "/home/nickv/ownCloud/Smashbox/smashbox/lib/owncloud/test_locking.py", line 106, in owner_worker
    test_var = lock_provider.is_locked(lock['lock'], config.oc_account_name, lock['path'])
  File "/home/nickv/ownCloud/Smashbox/smashbox/lib/owncloud/test_locking.py", line 157, in is_locked
    **kwargs
  File "/home/nickv/ownCloud/Smashbox/smashbox/src/pyocclient/owncloud/owncloud.py", line 1423, in make_ocs_request
    res = self.__make_ocs_request(method, service, action, **kwargs)
  File "/home/nickv/ownCloud/Smashbox/smashbox/src/pyocclient/owncloud/owncloud.py", line 1455, in __make_ocs_request
    res = self.__session.request(method, self.url + path, **attributes)
TypeError: request() got an unexpected keyword argument 'accepted_codes'

Comment (PVince81) [inline owncloud/owncloud.py]: it says `request()` not `make_ocs_request()`. Is that another method ?

Comment (nickvergessen) [inline owncloud/owncloud.py]: Well that's tunneled through from this call to `self.__make_ocs_request(method, service, action, **kwargs)` which itself calls then request()

Comment (nickvergessen) [inline owncloud/owncloud.py]: fixed, by using `pop()` instead of `get()`

**Comment by nickvergessen** [comment]
> Do you mind adding a test case as well ?

Mind explaining how that works and how I can run them locally?

**Comment by PVince81** [comment]
RTFM :wink: 
https://github.com/owncloud/pyocclient#running-the-unit-tests

The unit tests (which are more integration tests than unit tests) are here: https://github.com/owncloud/pyocclient/blob/master/owncloud/test/test.py#L81

**Comment by nickvergessen** [comment]
Done

**Comment by PVince81** [comment]
Fantastic :+1:

**Pull request merged**